packages/datadog-instrumentations/src/mocha.js

```diff
@@ -93,6 +99,38 @@ function isRetry (test) {
   return test._currentRetry !== undefined && test._currentRetry !== 0
 }
 
+function isNewTest (test) {
+  return !knownTests
+    .includes(`mocha.${getTestSuitePath(test.file, process.cwd())}.${test.fullTitle()}`)
+}
+
+function retryTest (test) {
+  const originalTestName = test.title
+  const suite = test.parent
+  for (let retryIndex = 0; retryIndex < earlyFlakeDetectionNumRetries; retryIndex++) {
+    const clonedTest = test.clone()
+    clonedTest.title = addEfdStringToTestName(originalTestName, retryIndex + 1)
+    suite.addTest(clonedTest)
+    clonedTest._ddIsNew = true
+    clonedTest._ddIsEfdRetry = true
+  }
+}
+
+function getFailingNewTests (root) {
+  let failingTests = []
```

Comment:
TODO: if _every_ attempt fails, we should still fail. It's just if it's flaky that we want non to block